packages/dev/core/src/ShadersWGSL/geometry.fragment.fx

```diff
@@ -100,8 +100,15 @@ fn main(input: FragmentInputs) -> FragmentOutputs {
             fragData[NORMAL_INDEX] =  vec4f(normalOutput, 1.0);
         #endif
     #else
-        fragData[0] =  vec4f(input.vViewPos.z / input.vViewPos.w, 0.0, 0.0, 1.0);
-        fragData[1] =  vec4f(normalOutput, 1.0);
+        #ifdef DEPTH
+            fragData[DEPTH_INDEX] = vec4f(input.vViewPos.z / input.vViewPos.w, 0.0, 0.0, 1.0);
+        #endif
+        #ifdef NORMAL
+            fragData[NORMAL_INDEX] = vec4f(normalOutput, 1.0);
+        #endif
+        #ifdef SCREENSPACE_DEPTH
+            fragData[SCREENSPACE_DEPTH_INDEX] = vec4f(fragmentInputs.position.z, 0.0, 0.0, 1.0);
+        #endif

```

Comment:
You will have to perform the same changes than in the WebGL case.

Reply:
Done